dist/metacom.js

```diff
@@ -62,7 +64,37 @@ export class Metacom extends EventEmitter {
     return new Transport(url, options);
   }
 
-  message(data) {
+  getStream(streamId) {
+    const stream = this.streams.get(streamId);
+    if (stream) return stream;
+    throw new Error(`Stream ${streamId} is not initialized`);
+  }
+
+  createStream(name, size) {
+    const streamId = ++this.streamId;
+    const initData = { streamId, name, size };
+    const transport = this;
+    return new MetacomWritable(transport, initData);
+  }
+
+  createBlobUploader(blob) {
+    const name = blob.name || 'blob';
+    const size = blob.size;
+    const consumer = this.createStream(name, size);
+    return {
+      streamId: consumer.streamId,
+      upload: async () => {
+        const reader = blob.stream().getReader();
+        let chunk;
+        while (!(chunk = await reader.read()).done) {
```

Comment:
We can use for..of here

Reply:
Unfortunatelly, browsers do not support stream async iterator contract, e.g. 
```js
ReadableStream.prototype[Symbol.asyncIterator]
```
while node.js does.